src/battle/quintohit.php

```diff
@@ -52,7 +52,8 @@
 			$db->execute("update `bixos` set `hp`=`hp`-? where `player_id`=?", [$totalpak, $player->id]);
 		}
 
-		$db->execute("update `players` set `mana`=`mana`-? where `id`=?", [$mana, $player->id]);
+		$player->mana -= $mana; //Correção de bug onde não reduzia a mana em um ataque quadruplo fatal.
+		// $db->execute("update `players` set `mana`=`mana`-? where `id`=?", [$mana, $player->id]);
```

Comment:
Not sure about this logic, you commented out

```php
$db->execute("update `players` set `mana`=`mana`-? where `id`=?", [$mana, $player->id]);
```

and changed it to

```php
$player->mana -= $mana;
```

which will only reduce the mana for the current player state, not in database.

The old core:
```php
$db->execute("update `players` set `mana`=`mana`-? where `id`=?", [$mana, $player->id]);
```

Was going the "most" correct thing, it was reducing the mana on database but not in the current `$player` state. Now, it's going to reduce the mana for the current state but not in database.

Reply:
It would actually be correct; however, the entire combat logic needs to be reviewed for the following reason. In other skills, when they are used, the player object loaded in the monster.php page is passed to these separate skill pages. Following the logic of other combat skills, it uses the received object itself and subtracts mana, so the update is done at the end of the turn on the monster.php page. Don’t ask me why this logic is structured this way, as I didn’t create it. I just applied what made sense according to the other skills! And I tested this adjustment several times, and it worked!

Reply:
Thank you for the explanation @Diguzz! LGTM.